### PR TITLE
Remove dependency between core and containers

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes https://github.com/kokkos/kokkos/issues/4109.

### CODE REVIEW

File: core/src/KokkosExp_InterOp.hpp [hunk 1]
```diff
@@ -100,16 +99,31 @@ using python_view_type_impl_t = typename python_view_type_impl<T...>::type;
...
+
+template <typename DataType, class... Properties>
+class DynRankView;
+
 namespace Experimental {
+
+// Duplicate from the header file for DynRankView to avoid core depending on
+// containers.
+template <class>
+struct is_dyn_rank_view : public std::false_type {};
```
Comment (dalg24) [inline core/src/KokkosExp_InterOp.hpp:111]: I would suffix with `_dupe` or similar to void possible confusion.

**Review by dalg24 — APPROVED** [review]

**Comment by masterleinad** [comment]
@ndellingwood Can you please confirm that this resolves #4109 for you?

**Review by crtrott — APPROVED** [review]

**Comment by ndellingwood** [comment]
@masterleinad I'm beginning testing with the previous failing build config

**Comment by ndellingwood** [comment]
@masterleinad I'm still seeing compilation errors with a fresh build:

```
[ 16%] Building CXX object kokkos/core/unit_test/CMakeFiles/KokkosCore_TestCompileOnly.dir/TestInterOp.cpp.o
cd /ascldap/users/ndellin/Trilinos-dev/Build/NightlyFail/kokkos/core/unit_test && /ascldap/users/ndellin/kokkos/bin/nvcc_wrapper   -I/ascldap/users/ndellin/Trilinos-dev/Build/NightlyFail -I/ascldap/users/ndellin/Trilinos-dev/kokkos/tpls/gtest -I/ascldap/users/ndellin/Trilinos-dev/Build/NightlyFail/kokkos/core/src -I/ascldap/users/ndellin/Trilinos-dev/kokkos/core/src -I/ascldap/users/ndellin/Trilinos-dev/Build/NightlyFail/kokkos -I/ascldap/users/ndellin/Trilinos-dev/Build/NightlyFail/kokkos/core/unit_test -I/ascldap/users/ndellin/Trilinos-dev/kokkos/core/unit_test -I/ascldap/users/ndellin/Trilinos-dev/kokkos/core/unit_test/category_files  -pedantic -Wall -Wno-long-long -Wwrite-strings    -arch=sm_70  -Xcudafe --diag_suppress=esa_on_defaulted_function_ignored -O3 -DNDEBUG   -std=c++14 -o CMakeFiles/KokkosCore_TestCompileOnly.dir/TestInterOp.cpp.o -c /ascldap/users/ndellin/Trilinos-dev/kokkos/core/unit_test/TestInterOp.cpp
/ascldap/users/ndellin/Trilinos-dev/kokkos/core/unit_test/TestInterOp.cpp:46:10: fatal error: Kokkos_DynRankView.hpp: No such file or directory
 #include <Kokkos_DynRankView.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [kokkos/core/unit_test/CMakeFiles/KokkosCore_TestCompileOnly.dir/TestInterOp.cpp.o] Error 1
```

**Comment by dalg24** [comment]
We need to conditionally enable the unit test

**Comment by masterleinad** [comment]
@ndellingwood Would you give it another shot?

**Review by ndellingwood — APPROVED** [review]
@masterleinad The recent update resolved the compilation errors, thanks!

**Pull request merged**